src/components/views/elements/AppTile.js

```diff
@@ -161,14 +161,18 @@ export default class AppTile extends React.Component {
         // if it's not remaining on screen, get rid of the PersistedElement container
         if (!ActiveWidgetStore.getWidgetPersistence(this.props.id)) {
             // FIXME: ActiveWidgetStore should probably worry about this?
-            const PersistedElement = sdk.getComponent("elements.PersistedElement");
-            PersistedElement.destroyElement(this._persistKey);
-            ActiveWidgetStore.delWidgetMessaging(this.props.id);
-            ActiveWidgetStore.delWidgetCapabilities(this.props.id);
-            ActiveWidgetStore.delRoomId(this.props.id);
+            this._destroyPersistentWidget();
         }
     }
 
+    _destroyPersistentWidget() {
```

Comment:
seems duplicated with ActiveWidgetStore::onRoomStateEvents. Have a single public method in ActiveWidgetStore that does this?

Reply:
Yeah, was on the fence about whether to do this now or leave it until moving more stuff to the store, but probably better to do it now.